packages/connect/src/protocol/universal-handler.ts

```diff
@@ -189,12 +209,33 @@ export function createUniversalMethodHandler(
  * different protocols - and returns a single handler that looks at the
  * Content-Type header and the HTTP verb of the incoming request to select
  * the appropriate protocol-specific handler.
+ *
+ * Raises an error if no protocol handlers were provided, or if they do not
+ * handle exactly the same RPC.
  */
-function negotiateProtocol(
-  service: ServiceType,
-  method: MethodInfo,
+export function negotiateProtocol(
   protocolHandlers: UniversalHandler[]
 ): UniversalHandler {
+  if (protocolHandlers.length == 0) {
+    throw new ConnectError("require at least one protocol", Code.Internal);
```

Comment:
maybe `at least one protocol is required`?

Reply:
TY, updated